bayesian/api_v1.py

```diff
@@ -832,33 +812,55 @@ def post():
                                           "manifest file '{filename}'".format(filename=filename))
 
             # Record the response details for this manifest file
+
             manifest = {'filename': filename,
                         'content': content,
                         'ecosystem': ecosystem or manifest_descriptor.ecosystem,
                         'filepath': filepath}
 
             manifests.append(manifest)
 
+            if not ecosystem:
+                ecosystem = get_ecosystem_from_manifest(manifests)
+
         data = {'api_name': 'stack_analyses'}
         args = {'external_request_id': request_id,
                 'ecosystem': ecosystem, 'data': data}
         try:
             api_url = current_app.config['F8_API_BACKBONE_HOST']
-
             d = DependencyFinder()
-            if origin == "vscode" and ecosystem in ("npm", "pypi"):
+            deps = {}
+            worker_flow_enabled = "false"
+
+            if resolved_files_exist(manifests) == "true" \
+                    and ecosystem in ("npm", "pypi"):
                 deps = d.scan_and_find_dependencies(ecosystem, manifests)
+            elif scan_repo_url and ecosystem == "npm":
+
```

Comment:
All these conditions should mention what it is doing
e.g. this elif is we get a git url . Add TODO that says, add support for other ecosystems as well. Currently it only supports npm